### PR TITLE
fix: Align fnv1a32 hash function with JS implementation

### PR DESCRIPTION
**Pull request by vazarkevych**
Corrects the FNV-1a hash implementation in Dart to precisely match the behavior of the original JavaScript function

### CODE REVIEW

File: lib/src/Utils/gb_utils.dart [hunk 1]
```diff
@@ -20,24 +20,18 @@ class FNV {
...
-            (hash << 24) +
-            (hash << 8) +
-            (hash << 7) +
-            (hash << 4) +
-            (hash << 1) +
-            hash
-        ).toUnsigned(32); 
-    }
-    return hash;
+  int fnv1a32(String str) {
```
Comment (madhuchavva) [inline lib/src/Utils/gb_utils.dart:23]: This is a high-impact correctness fix because hash parity affects experiment assignment. Before merging, can we add official cross-SDK test vectors for `fnv1a32` and assignment outputs? The implementation looks intended to match JS overflow behavior, but without fixed vectors it’s hard to verify Dart signed/unsigned transitions for representative inputs, unicode/UTF-16 cases, and known GrowthBook hash examples. A small table of expected hashes from the JS SDK would make this safe to approve and protect against future regressions.

**Review by madhuchavva — COMMENTED** [review]
Inline review comment requesting official hash parity vectors.

**Review by madhuchavva — CHANGES_REQUESTED** [review]
Requesting changes based on the inline review comments. The notes call out functional gaps or compatibility/documentation risks that should be addressed before merge.

File: lib/src/Utils/gb_utils.dart (hunk 1 above)
Comment (vazarkevych) [reply lib/src/Utils/gb_utils.dart:23]: Done — added in new [PR](https://github.com/growthbook/growthbook-flutter/pull/158). The test file covers ASCII vectors, Unicode/UTF-16 inputs, and v2 double-hash intermediates, all verified against the JS SDK's hashFnv32a().
